### PR TITLE
fix: do not throw errors but catch and log them

### PR DESCRIPTION
**Pull request by marcklingen**
Goal:
- in cases where the SDK currently throws errors, they should only console.error()
- Add tests for edge cases to check that the SDK actually catches those errors and does not throw into global context

### CODE REVIEW

File: langfuse-core/src/index.ts [hunk 1]
```diff
@@ -394,14 +394,12 @@ abstract class LangfuseCoreStateless {
       body: payload,
     });
 
-    const requestPromise = this.fetchWithRetry(url, fetchOptions);
-    this.pendingPromises[promiseUUID] = requestPromise;
```
Comment (marcklingen) [inline langfuse-core/src/index.ts:398]: No uncatched promises in pendingPromises

File: langfuse-core/src/index.ts [hunk 2]
```diff
@@ -496,9 +494,6 @@ abstract class LangfuseCoreStateless {
         )
       );
     } catch (e) {
-      if (!isLangfuseFetchError(e)) {
```
Comment (marcklingen) [inline langfuse-core/src/index.ts:499]: Do not throw any errors

**Comment by marcklingen** [comment]
These are improvements, but until now core issue is not reproduced/fixed, the current test suite of this pr also passes on main

File: integration-test/langfuse-integration-error-handling.spec.ts [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+      const trace = langfuse.trace({ name: "trace-name" });
+      for (let i = 0; i < 10; i++) {
+        trace.generation({ name: "generation-name" });
+      }
+
+      await new Promise((resolve) => setTimeout(resolve, 4000));
+      await langfuse.shutdownAsync();
+
+      // expect no errors to be thrown (would kill jest) and console.error to be called
+      expect(true).toBe(true);
```
Comment (maxdeichmann) [inline integration-test/langfuse-integration-error-handling.spec.ts:36]: expect(true).toBe(true); is not necessary here i think

**Comment by maxdeichmann** [comment]
- [enqueue](https://github.com/langfuse/langfuse-js/blob/5cd24ecb99106368eec611579ff95084601e06cc/langfuse-core/src/index.ts#L317) is not wrapped in try catch block, neither are all the user facing functions like `trace()`. `flush()` might be called in `enqueue`. If there is an exception thrown, this will throw exceptions into our users code
- Is it possible to mock `fetchWithRetry` and let it fail? I see that you added tests which will fail but this would be more explicit.
- [flushAsync](https://github.com/langfuse/langfuse-js/blob/5cd24ecb99106368eec611579ff95084601e06cc/langfuse-core/src/index.ts#L344) rejects the promise which we return to the user. Is this intended?

File: integration-test/langfuse-integration-error-handling.spec.ts (hunk 3 above)
Comment (marcklingen) [reply integration-test/langfuse-integration-error-handling.spec.ts:36]: yes, initially only had this test, now also have another one. removing this.

Comment (marcklingen) [reply integration-test/langfuse-integration-error-handling.spec.ts]: agree, removed

File: integration-test/langfuse-integration-error-handling.spec.ts [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+import { FakeListLLM } from "langchain/llms/fake";
+
+import { LANGFUSE_BASEURL, LANGFUSE_PUBLIC_KEY, LANGFUSE_SECRET_KEY } from "./integration-utils";
+
+describe("No errors should be thrown by SDKs", () => {
+  jest.useRealTimers();
+
+  // beforeEach(() => {});
+  // afterEach(async () => {});
```
Comment (maxdeichmann) [inline integration-test/langfuse-integration-error-handling.spec.ts:13]: comment

File: integration-test/langfuse-integration-error-handling.spec.ts [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+      // expect no errors to be thrown (would kill jest)
+      expect(global.console.error).toHaveBeenCalledTimes(1);
+    }, 10000);
+  });
+});
+
+describe("shutdown async behavior", () => {
+  jest.useRealTimers();
+
+  // beforeEach(() => {});
```
Comment (maxdeichmann) [inline integration-test/langfuse-integration-error-handling.spec.ts:120]: comment
